File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.HashSet;

class Solution {
    public int solution(int[] nums) {
        int answer = 0;
        int arrsize = nums.length / 2;
        HashSet<Integer> mp = new HashSet<>();
        for (int num : nums){
            mp.add(num);
        }
        if (mp.size() >= arrsize) answer = arrsize;
        else answer = mp.size();
        return answer;
    }
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.cpp
#include <vector>
#include <map>

using namespace std;

int solution(vector<int> nums)
{
    int answer = nums.size() / 2;
    map<int,int> counts;
    
    for (int num : nums){
        counts[num]++;
    }
    int temp =0;
    for (auto a : counts){
        temp++;
    }
    answer = answer < temp ? answer : temp;
    return answer;
}